### PR TITLE
Added Least squares implementation

### PR DESCRIPTION
**Pull request by jordanmontt**
Thanks to @olekscode work, we propose to have the least squares implementation in PolyMath.

This PS adds two classes: PMLeastSquares and its tests. 

"SVD-based implementation of a minimum-norm solution to the overdetermined least squares problem."

### CODE REVIEW

File: src/Math-Numerical/PMLeastSquares.class.st [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+SVD-based implementation of a minimum-norm solution to the overdetermined least squares problem.
+"
+Class {
+	#name : #PMLeastSquares,
+	#superclass : #Object,
+	#category : #'Math-Numerical-Math-LinearAlgebra'
+}
+
+{ #category : #accessing }
+PMLeastSquares >> pseudoinverseOfDiagonal: aMatrix [
```
Comment (hemalvarambhia) [inline src/Math-Numerical/PMLeastSquares.class.st:11]: Brilliant naming!

File: src/Math-Tests-Numerical/PMLeastSquaresTest.class.st [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+	super setUp.
+	leastSquares := PMLeastSquares new.
+]
+
+{ #category : #tests }
+PMLeastSquaresTest >> testPseudoinverseOfDiagonalSquareMatrix [
+
+	| matrix expectedInverse inverse |
+	
+	matrix := PMMatrix rows: #(
```
Comment (hemalvarambhia) [inline src/Math-Tests-Numerical/PMLeastSquaresTest.class.st:24]: Consider consistency in formatting the rows

**Review by hemalvarambhia — APPROVED** [review]
Brilliant

File: src/Math-Tests-Numerical/PMLeastSquaresTest.class.st (hunk 2 above)
Comment (jordanmontt) [reply src/Math-Tests-Numerical/PMLeastSquaresTest.class.st:25]: I did a new commit formatting the rows

**Review by hemalvarambhia — APPROVED** [review]

**Comment by SergeStinckwich** [comment]
Congrats for your PR!

**Pull request merged**